packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -248,16 +260,43 @@ class InMemoryAnchorService implements AnchorService {
     } else {
       this.#queue.push(candidate);
     }
-    return concat(
-      of<AnchorServiceResponse>({
-        status: AnchorStatus.PENDING,
-        streamId: streamId,
-        cid: tip,
-        message: "Sending anchoring request",
-        anchorScheduledFor: null,
-      }),
-      feed$
+    this._emitResponse({
+      status: AnchorStatus.PENDING,
+      streamId: streamId,
+      cid: tip,
+      message: "Sending anchoring request",
+      anchorScheduledFor: null,
+    })
+    return this.pollForAnchorResponse(streamId, tip)
+  }
+
+  /**
+   * Start polling the anchor service to learn of the results of an existing anchor request for the
+   * given tip for the given stream.
+   * @param streamId - Stream ID
+   * @param tip - Tip CID of the stream
+   */
+  pollForAnchorResponse(streamId: StreamID, tip: CID): Observable<AnchorServiceResponse> {
+    const anchorResponse = this.#anchors.get(tip.toString())
+    const feed$ = this.#feed.pipe(
+      filter((asr) => asr.streamId.equals(streamId) && asr.cid.equals(tip))
     );
+
+    if (anchorResponse) {
+      return concat(
+        of<AnchorServiceResponse>(anchorResponse),
+        feed$
+      );
+    } else {
+      return concat(
+        of<AnchorServiceResponse>({
+          status: AnchorStatus.FAILED,
+          streamId,
+          cid: tip,
+          message: "Request not found",
+        }),
+        feed$);
```

Comment:
If it is FAILED, why bother returning something else?

Reply:
You're asking why there's still the `concat()` with `feed$`?  I actually originally had that removed but I put it back while I was trying to debug the issue I mentioned in [this comment](https://github.com/ceramicnetwork/js-ceramic/pull/1437/files#r643462567).  Since it didn't help I can remove it